backend/src/server_process_helper.py

```diff
@@ -59,7 +63,25 @@ def _read_output(self):
         for line in self._process.stdout:
             if self._stop_event.is_set():
                 break
-            print(line.decode().strip())
+            stripped_line = line.decode().strip()
+            match_obj = re.match(SANIC_LOG_REGEX, stripped_line)
```

Comment:
You are saying you are concerned about performance, yet you are compiling a regex in every iteration of the loop. Curious.

Jokes aside, please don't do that. Compile the re outside the loop and use it inside.